### PR TITLE
Android: Address bar suggestions keep changing

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/414730916066338/1208634701288221/f

### Description
Fixed suggestions changing issue.

### Steps to test this PR

To reproduce the issue:
- [x] Install the latest nightly build
- [x] Set your device language to Hungarian (Magyar)
- [x] Open a tab and start searching for something.
- [x] Notice the suggestions keep changing. See also the video attached to the Asana task.

To test the fix:
- [x] Install from this branch
- [x] Set your device language to Hungarian (Magyar)
- [x] Open a tab and start searching for something.
- [x] Notice the suggestions don't change once they're displayed.

### NO UI changes

### CODE REVIEW

**Comment by anikiki** [comment]
* **#5215** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5215?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @anikiki and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt [hunk 1]
```diff
@@ -1484,8 +1484,7 @@ class BrowserTabViewModelTest {
             whenever(mockAutoCompleteScorer.score("title", "https://foo.com".toUri(), 1, "title")).thenReturn(1)
             whenever(mockUserStageStore.getUserAppStage()).thenReturn(ESTABLISHED)
 
-            testee.onAutoCompleteSuggestionsChanged()
-            testee.autoCompleteStateFlow.value = "title"
+            testee.triggerAutocomplete("title", hasFocus = true, hasQueryChanged = true)
```
Comment (anikiki) [inline app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt:1487]: This has changed as the `onAutoCompleteSuggestionsChanged` was removed.

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt [hunk 2]
```diff
@@ -1745,7 +1746,9 @@ class BrowserTabFragment :
             }
             is Command.SwitchToTab -> {
                 binding.focusedView.gone()
-                viewModel.autoCompleteSuggestionsGone()
+                if (binding.autoCompleteSuggestionsList.isVisible) {
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:1749]: Added this condition everywhere to ensure we don't send any pixels (this happens in `autoCompleteSuggestionsGone()`) if the `autoCompleteSuggestionsList` is hidden more than once (noticed some edge-cases).

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt [hunk 3]
```diff
@@ -755,8 +756,7 @@ class BrowserTabViewModel @Inject constructor(
     @OptIn(FlowPreview::class, ExperimentalCoroutinesApi::class)
     @SuppressLint("CheckResult")
     private fun configureAutoComplete() {
-        autoCompleteJob?.cancel()
-        autoCompleteJob = autoCompleteStateFlow
+        autoCompleteJob += autoCompleteStateFlow
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt:759]: Updated to use the `ConflatedJob` to simplify the code.

**Review by CrisBarreiro — APPROVED** [review]
works as expected, great work!

**Pull request merged**